tests/units/array/test_array_from_to_bytes.py

```diff
@@ -69,3 +69,17 @@ def test_from_to_base64(protocol, compress, show_progress):
         assert d1.image.url == d2.image.url
     assert da[1].image.url is None
     assert da2[1].image.url is None
+
+
+def test_union_type_error(tmp_path):
+    from typing import Union
+
+    from docarray.documents import TextDoc
+
+    class CustomDoc(BaseDoc):
```

Comment:
what about Union[int, float] or Union[int, str]

Reply:
After the testing, the Union with basis types should work in all serialization methods. Only the Union with document types will cause those issues.